worldwind/src/commonMain/kotlin/earth/worldwind/render/buffer/StorageBufferPool.kt

```diff
@@ -0,0 +1,121 @@
+package earth.worldwind.render.buffer
+
+import earth.worldwind.util.kgl.KglBuffer
+import earth.worldwind.draw.DrawContext
+
+val DEFAULT_BUFFER_POOL_BLOCK_SIZE : Int = 4 * 1024
+val DEFAULT_BUFFER_POOL_MAX_BLOCK_COUNT : Int = 64
+class StorageBufferPool(private var target : Int,
+        private var usage : Int,
+        private var blockSize : Int = DEFAULT_BUFFER_POOL_BLOCK_SIZE,
+        private var blockCount : Int = DEFAULT_BUFFER_POOL_MAX_BLOCK_COUNT)
+{
+    private class StorageBufferObject() {
```

Comment:
IN general you can avoid using this class. size is known in pool, offset is only used for current buffer, and only Buffer is useful field.

Reply:
done